File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        s=set(nums)
        maxi=0
        for i in s:
            if i>0:
                maxi=max(maxi,i)
        for i in range(1,maxi+1):
            if i not in s:
                return i
        return maxi+1
        

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Map<Integer,Boolean> map=new HashMap<>();
        long maxi=0;
        for(int i:nums){
            if(i<=0){
                continue;
            }else{
                map.put(i,true);
                maxi=Math.max(maxi,i);
            }
        }
        for(int i=1;i<maxi+1;i++){
            if(!map.containsKey(i)){
                return i;
            }
        }
        return (int)maxi+1;
    }
}